### PR TITLE
Make uncaught errors in auto-booted containers fatal

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: system/extensions/esp32/boot.toit [hunk 1]
```diff
@@ -50,5 +50,11 @@ main:
   ]
   container_manager.register_system_image
       SystemImage container_manager
-  boot container_manager
-  // TODO(kasper): Should we reboot here after a little while?
+
+  error ::= boot container_manager
```
Comment (floitsch) [inline system/extensions/esp32/boot.toit:54]:
```suggestion
  exit_code ::= boot container_manager
```

**Review by floitsch — APPROVED** [review]
LGTM.

File: system/containers.toit [hunk 2]
```diff
@@ -276,17 +269,18 @@ class ContainerManager extends ContainerServiceDefinition implements SystemMessa
...
+  on_container_stop_ container/Container error/int -> none:
     containers_by_id_.remove container.id
+    // If we've got an error in an automatically booted image, we treat that
+    // as a fatal error and terminate the system process.
+    if error != 0 and container.image.run_on_boot:
+      done_.set error
+      return
     // TODO(kasper): We are supposed to always have a running system process. Maybe
     // we can generalize this handling and support background processes that do not
     // restrict us from exiting?
```
[the remark this answers is not in the document]
Comment (kasperl) [reply system/containers.toit:286]: It still applies :)

File: system/containers.toit [hunk 3]
```diff
@@ -44,26 +44,24 @@ class Container:
...
   on_stop_ -> none:
-    pids_.do: on_process_stop_ it
+    pids_.do: on_process_stop_ it 0
 
   on_process_start_ pid/int -> none:
     assert: not pids_.is_empty
     pids_.add pid
 
-  on_process_stop_ pid/int -> none:
+  on_process_stop_ pid/int error/int -> none:
```
[the remark this answers is not in the document]
Comment (kasperl) [reply system/containers.toit:57]: Might fix this in one go. There are more occurrences.

**Pull request merged**